### ISSUE TITLE
Vim syntax file / plugin for tridactylrc

### ISSUE THREAD

**Issue by bovine3dom** [cmcaine/tridactyl#589]
Should probably be a separate repo to make installation easy with vim-plug et al.

Nice snippets to extract the valid excommands for partial automation:
```bash
grep -E "function.*{" excmds.ts | sed "s/.*function //" | sed "s/(.*//" | sort | uniq
```

```bash
sed "/exaliases/,/}/!d" config.ts | sed "s/:.*//g" | tail -n +2 | head -n -1 | awk '{$1=$1};1' | sed 's/"//g'
```

**Comment by glacambre** [comment]
I don't think we can do excmd validation in the syntax file since users can define their own excmds.

Also, Pentadactyl had a [command to generate runtime files](https://github.com/5digits/dactyl/blob/0043f8dbb629a8e8836241c4c53411afc8cba8de/common/locale/en-US/various.xml#L76). I think I'd prefer a separate repo though.

**Comment by bovine3dom** [comment]
> I don't think we can do excmd validation in the syntax file since users can define their own excmds.

Yeah, but we can give ours a nice highlight and maybe warn people that inline comments aren't allowed, etc.

**Comment by bovine3dom** [comment]
Prior art:

https://github.com/joedicastro/vim-pentadactyl
https://github.com/5digits/dactyl/blob/master/common/javascript.vim

**Comment by alerque** [comment]
This is something I might be able to contribute towards a bit. I also think a separate repo is probably the way to go. My only hesitation is a thought playing around the back of my mind that someday Tidactyl's in-browser command line might have syntax highlighting as well, in which case it might be nice to somehow share code ... but perhaps the only thing shared would be generated lists of keywords. 
In any case starting out in a dedicated repository is probably the right way to start. Trying to bootstrap a new layout on this repository so it would work as both a vim plugin _and_ it's current role (ala Firenvim style) probably isn't work the mess it will make.

To that end if somebody would like to fire up a repository in this Github org for `vim-tridactyl` and give me some access to it I'll give it a go. Or if you'd rather I start a repository in my own namespace for now and import it later when it meets muster that's okay too, just let me know what you'd all prefer.

**Comment by alerque** [comment]
I've actually gone ahead and started coding for this and pushed it to a repo in my namespace at [`alerque/vim-tridactyl`](https://github.com/alerque/vim-tridactyl), but the readme, CI lint status job, etc. are coded up as if the canonical source repository is `tridactyl/vim-tridactyl` under the assumption that it will just get pushed to this org and I'll drop my copy. When I hear back and how you guys would prefer to manage this I can rebase to set the paths correctly if that assumption is not correct.

**Comment by bovine3dom** [comment]
I've forked it to this organisation and added you as a collaborator :) GitHub makes those invitations hard to find - check your emails.

Once you're happy with it it would be nice to add a `:mkv` command which installs it from Tridactyl if you have the native messenger.

**Comment by alerque** [comment]
Thanks @bovine3dom that's great. I just nuked the repo from my namespace to clear the "fork" status (so the org one is canonical). I think a residual effect of the fork process is that the Github "issues" feature is disabled. Since I don't have access to the settings for the repository perhaps you can enable that.

**Comment by bovine3dom** [comment]
Ah, I see. Done!

**Comment by bovine3dom** [comment]
https://github.com/tridactyl/vim-tridactyl (although now so old that it could do with updating)
